static/app/views/app/index.tsx

```diff
@@ -35,201 +31,151 @@ const GlobalNotifications = HookOrDefault({
   defaultComponent: () => null,
 });
 
-function getAlertTypeForProblem(problem) {
-  switch (problem.severity) {
-    case 'critical':
-      return 'error';
-    default:
-      return 'warning';
-  }
-}
-
 type Props = {
-  api: Client;
   config: Config;
-} & RouteComponentProps<{}, {}>;
-
-type State = {
-  loading: boolean;
-  error: boolean;
-  needsUpgrade: boolean;
-  newsletterConsentPrompt: boolean;
-  user?: Config['user'];
-  organization?: Organization;
+  children: React.ReactNode;
 };
 
-class App extends Component<Props, State> {
-  state: State = {
-    loading: false,
-    error: false,
-    needsUpgrade: ConfigStore.get('user')?.isSuperuser && ConfigStore.get('needsUpgrade'),
-    newsletterConsentPrompt: ConfigStore.get('user')?.flags?.newsletter_consent_prompt,
```

Comment:
We no longer store either of these in state, since the component receives updates from using `withConfig`.